addons/mail/static/src/models/message_tracking_value.js

```diff
@@ -0,0 +1,182 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+
+import { format } from 'web.field_utils';
+
+registerModel({
+    name: 'MessageTrackingValue',
+    identifyingFields: ['id'],
+    modelMethods: {
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+        /**
+         * @static
+         * @param {Object} data
+         * @return {Object}
+         */
+        convertData(data) {
+            const data2 = {};
+            if ('id' in data) {
+                data2.id = data.id;
+            }
+            if ('changed_field' in data) {
+                data2.changedField = data.changed_field;
+            }
+            if ('currency_id' in data) {
+                data2.currencyId = data.currency_id;
+            }
+            if ('field_type' in data) {
+                data2.fieldType = data.field_type;
+            }
+            if ('new_value' in data) {
+                data2.newValue = data.new_value;
+            }
+            if ('old_value' in data) {
+                data2.oldValue = data.old_value;
+            }
+            return data2;
+        },
+
+    },
+    recordMethods: {
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeChangedFieldAsString() {
+            return _.str.sprintf(this.env._t("%s:"), this.changedField);
+        },
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeNewValueAsString() {
+           return this._formatMessageTrackingValue(this.fieldType, this.newValue) || this.env._t("None");
+       },
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeOldValueAsString() {
+            return this._formatMessageTrackingValue(this.fieldType, this.oldValue) || this.env._t("None");
+        },
+
+        /**
+         * @private
+         * @param {string} fieldType
+         * @param {string|boolean|float|number|integer|Array|Object} value
+         * @returns {string}
+         */
+        _formatMessageTrackingValue(fieldType, value) {
```

Comment:
The existence of this method and the need to pass "value" as param shows we are missing a generic "value" model (the same model containing both old and new), and that old value and new value should be relations to it on a specific "tracking item" model.

Note: the python model is "wrong" in this regard, so we might need to split `mail.tracking.value` model (the source of the server) from newly computed "tracking item" and "tracking value" models in JS.